### PR TITLE
[radio] refine the definition of SYMBOL_TIME

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR simplied the definition of `OT_RADIO_SYMBOL_TIME` and
provided the reference of the value from IEEE 802.15.4-2015.

### CODE REVIEW

File: include/openthread/platform/radio.h [hunk 1]
```diff
@@ -68,16 +68,18 @@ extern "C" {
...
+
+    // Per IEEE 802.15.4-2015, 12.3.3 Symbol rate:
+    // The O-QPSK PHY symbol rate shall be 25 ksymbol/s when operating in the 868 MHz band and 62.5 ksymbol/s when
+    // operating in the 780 MHz, 915 MHz, 2380 MHz, or 2450 MHz band
+    OT_RADIO_SYMBOL_RATE = 62500, ///< The O-QPSK PHY symbol rate when operating in the 780MHz, 915MHz, 2380MHz, 2450MHz
+    OT_RADIO_SYMBOL_TIME_US      = 1000000 * 1 / OT_RADIO_SYMBOL_RATE, ///< Symbol duration time
+    OT_RADIO_TEN_SYMBOLS_TIME_US = 10 * OT_RADIO_SYMBOL_TIME_US,       ///< Time for 10 symbols
+    OT_RADIO_LQI_NONE            = 0,                                  ///< LQI measurement not supported
+    OT_RADIO_RSSI_INVALID        = 127,                                ///< Invalid or unknown RSSI value
+    OT_RADIO_POWER_INVALID       = 127,                                ///< Invalid or unknown power value
```
Comment (abtink) [inline include/openthread/platform/radio.h:82]: Same here. I think we should keep the old names in `radio.h` platform header (platform implementations may be using them already). `OT_RADIO_SYMBOLS_PER_OCTET` and  `OT_RADIO_BIT_RATE` and `OT_RADIO_BITS_PER_OCTET`


We can add new definitions?

File: include/openthread/link.h [hunk 2]
```diff
@@ -52,7 +52,6 @@ extern "C" {
  * @{
  *
  */
-#define OT_US_PER_TEN_SYMBOLS 160 ///< The microseconds per 10 symbols.
```
Comment (abtink) [inline include/openthread/link.h:55]: Suggest maybe keeping this one as is?

In general, we should try to keep definitions in public or platform headers the same as much as we can (to have backward compatibility with code that may be using OT or implementations of platform layer). 

We can change a public API or definition if we have a good justification for the change (but better to limit such changes).

File: src/core/radio/radio.hpp [hunk 3]
```diff
@@ -45,7 +45,7 @@
 
 namespace ot {
 
-static constexpr uint32_t kUsPerTenSymbols = OT_US_PER_TEN_SYMBOLS; ///< The microseconds per 10 symbols.
+static constexpr uint32_t kUsPerTenSymbols = OT_RADIO_TEN_SYMBOLS_TIME_US; ///< The microseconds per 10 symbols.
```
Comment (abtink) [inline src/core/radio/radio.hpp:48]: Change the internal name `kUsPerTenSymbols`?

File: include/openthread/link.h (hunk 2 above)
Comment (Irving-cl) [reply include/openthread/link.h:55]: Make sense to me. Let's keep all these existing definitions for compatibility. In this PR I just added the definition of `SYMBOL_RATE`.

Actually the major intention of this PR is to add explanation for how 160 comes from. Even I forgot this for a few times...

File: src/core/radio/radio.hpp (hunk 3 above)
Comment (Irving-cl) [reply src/core/radio/radio.hpp]: Let's keep this unchanged since we will keep `OT_US_PER_TEN_SYMBOLS`.

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

**Review by jwhui — APPROVED** [review]

**Pull request merged**